internal/build/resolver/current_branch.go

```diff
@@ -13,6 +13,12 @@ import (
 
 // ResolveBuildFromCurrentBranch Finds the most recent build for the branch in the current working directory
 func ResolveBuildFromCurrentBranch(repo *git.Repository, pipelineResolver pipelineResolver.PipelineResolverFn, f *factory.Factory) BuildResolverFn {
+	// there is nothing to resolve from if we aren't in a git repository, so short circuit that
+	if repo == nil {
+		return func(ctx context.Context) (*build.Build, error) {
+			return nil, nil
+		}
+	}
```

Comment:
This is the first option. We would include this resolver every time, but it will check for a nil repo which means not in a repository, and return a dummy resolver to pass through to the next one.

Reply:
This approach handles the error at the right time which I think should be the right thing to do.